src/bootstrap/tool.rs

```diff
@@ -214,10 +212,17 @@ impl Step for ToolBuild {
             if tool == "tidy" {
                 tool = "rust-tidy";
             }
-            let cargo_out =
-                builder.cargo_out(compiler, self.mode, target).join(exe(tool, compiler.host));
-            let bin = builder.tools_dir(compiler).join(exe(tool, compiler.host));
+            let exe = exe(tool, compiler.host);
+            let cargo_out = builder.cargo_out(compiler, self.mode, target).join(&exe);
+            let bin = builder.tools_dir(compiler).join(&exe);
             builder.copy(&cargo_out, &bin);
+
+            // Don't create a stage0-sysroot/bin directory.
```

Comment:
Doing this for all tools is likely not a good idea, as it'll place things like tidy and other internal tools in the PATH (particularly on windows, that may be a problem). It should be possible to add this after the calls to ToolBuild, as we return the appropriate path already.